gitclone/checkout.go

```diff
@@ -53,49 +59,100 @@ type checkoutStrategy interface {
-// | tag         |        |  X !|        |          |          |           |          |
-// | branch      |  _     |  _  |  X !   |  X       |  X       |  X        |          |
-// | branchDest  |        |     |        |  X       |  X       |           |  X       |
-// | PRRepoURL   |        |     |        |      !   |  X !     |    !      |    !     |
-// | PRID        |        |     |        |          |          |           |    !     |
-// | mergeBranch |        |     |        |          |          |  X !      |          |
-// |==================================================================================|
-
-func selectCheckoutMethod(cfg Config) CheckoutMethod {
-	isPR := cfg.PRRepositoryURL != "" || cfg.PRMergeBranch != "" || cfg.PRID != 0
+// |==========================================================================================================================|
+// | params\strat| commit | tag | branch | manualMR | manualPR | autoMerge | autoDiff | noMergeHeadBranch | noMergeForkBranch |
+// | commit      |  X  !  |     |        |  X       |          |           |          |                   |                   |
+// | tag         |        |  X !|        |          |          |           |          |                   |                   |
+// | branch      |  _     |  _  |  X !   |  X       |  X       |  X        |          |                   |  X                |
+// | branchDest  |        |     |        |  X       |  X       |           |  X       |                   |                   |
+// | PRRepoURL   |        |     |        |      !   |  X !     |    !      |    !     |                   |  X !              |
+// | PRID        |        |     |        |          |          |           |    !     |                   |    !              |
+// | mergeBranch |        |     |        |          |          |  X !      |          |    !              |                   |
+// | headBranch  |        |     |        |          |          |           |          |  X !              |                   |
+// |==========================================================================================================================|
+
+func selectCheckoutMethod(cfg Config, patch patchSource) (CheckoutMethod, string) {
+	isPR := cfg.PRRepositoryURL != "" || cfg.BranchDest != "" || cfg.PRMergeBranch != "" || cfg.PRID != 0
 	if !isPR {
 		if cfg.Commit != "" {
-			return CheckoutCommitMethod
+			return CheckoutCommitMethod, ""
 		}
 
 		if cfg.Tag != "" {
-			return CheckoutTagMethod
+			return CheckoutTagMethod, ""
 		}
 
 		if cfg.Branch != "" {
-			return CheckoutBranchMethod
+			return CheckoutBranchMethod, ""
 		}
 
-		return CheckoutNoneMethod
+		return CheckoutNoneMethod, ""
 	}
 
 	isFork := isFork(cfg.RepositoryURL, cfg.PRRepositoryURL)
-	isPrivateFork := isPrivate(cfg.PRRepositoryURL) && isFork
+	isPrivateSourceRepo := isPrivate(cfg.PRRepositoryURL)
+	isPrivateFork := isFork && isPrivateSourceRepo
```

Comment:
It would be great to consolidate the naming.
We use `source` and `head` for PR head properties (repo, branch, etc)

Reply:
Updated every usage to source/destination.